### PR TITLE
Mixed operator algebra methods for fermion ops and pauli ops.

### PR DESCRIPTION
**Pull request by kharazit**
Need to figure out how to use Observables to generalize and simply some of this a little

### CODE REVIEW

File: examples/fermi/mixed_op.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    std::cout<<"mixed product "<<std::endl;
+
+    auto mixed_prod = adag(0)*X(1) + X(0)*a(1);
+
+    std::cout << mixed_prod.toString() << std::endl;
+
+    auto H = X(0) + X(0)*a(1) + adag(1)*X(0) + X(0)*adag(0)*adag(1);
+
+    std::cout<<"ARE WE HERE??"<<std::endl;
```
Comment (amccaskey) [inline examples/fermi/mixed_op.cpp:42]: we can remove this one I think

File: runtime/CMakeLists.txt [hunk 2]
```diff
@@ -6,7 +6,11 @@ add_library(${LIBRARY_NAME} SHARED ${SRC})
 
 target_include_directories(${LIBRARY_NAME} PUBLIC . qrt)
 
+<<<<<<< HEAD
 target_link_libraries(${LIBRARY_NAME} PUBLIC xacc::xacc xacc::quantum_gate qrt xacc::pauli)
+=======
+target_link_libraries(${LIBRARY_NAME} PUBLIC xacc::xacc xacc::quantum_gate qrt xacc::pauli xacc::fermion)
```
Comment (amccaskey) [inline runtime/CMakeLists.txt:12]: need to remove the auto-generated merge strings, just keep this target_link_libraries call

File: runtime/qcor.cpp [hunk 3]
```diff
@@ -62,7 +61,15 @@ double observe(std::shared_ptr<CompositeInstruction> program, Observable &obs,
 }
 } // namespace __internal__
 
-std::shared_ptr<xacc::Optimizer> createOptimizer(const std::string &type,
+std::shared_ptr<Observable> transform(Observable &obs, std::string transf){
+  if (transf != "jw"){
```
Comment (amccaskey) [inline runtime/qcor.cpp:65]: Let's update this to not have this if statement (and we don't want to throw), and line 69 should just take transf. Let the xacc service registry throw an error if the string is not jw

File: runtime/qcor.hpp [hunk 4]
```diff
@@ -25,46 +27,155 @@ using Observable = xacc::Observable;
 using Optimizer = xacc::Optimizer;
 using CompositeInstruction = xacc::CompositeInstruction;
 using PauliOperator = xacc::quantum::PauliOperator;
+using FermionOperator = xacc::quantum::FermionOperator;
+
+PauliOperator X(int idx){
```
Comment (amccaskey) [inline runtime/qcor.hpp:32]: I think we should pull all the PauliOperator / FermionOperator stuff out into its own file. We should create runtime/utils/operator_algebra.hpp and add this stuff there and then just include it here in qcor.hpp

File: runtime/qcor.hpp [hunk 5]
```diff
@@ -295,17 +408,20 @@ auto observe(QuantumKernel &kernel, Observable &obs, Args... args) {
...
 std::shared_ptr<ObjectiveFunction> createObjectiveFunction(
-    const std::string &obj_name, std::shared_ptr<CompositeInstruction> kernel,
-    std::shared_ptr<Observable> observable, HeterogeneousMap &&options = {}) {
-  auto obj_func = qcor::__internal__::get_objective(obj_name);
-  obj_func->initialize(observable.get(), kernel);
-  obj_func->set_options(options);
+    const std::string & obj_name, std::shared_ptr<CompositeInstruction> kernel,
+    std::shared_ptr<Observable> observable, HeterogeneousMap &&options = {}) 
+    {
+      std::cout<<"shared_ptr OBSERVABLE CALLED"<<std::endl;
```
Comment (amccaskey) [inline runtime/qcor.hpp:421]: don't need this

File: tools/driver/qcor.in [hunk 6]
```diff
@@ -8,7 +8,11 @@ def main(argv=None):
                             '-L', '@CMAKE_INSTALL_PREFIX@/lib', '-lxacc',
                             '-lqcor', '-lqrt', '-lCppMicroServices',
                             '-lxacc-quantum-gate',
+<<<<<<< HEAD
                             '-lxacc-pauli', '-lpthread']
+=======
+                            '-lxacc-pauli', '-lpthread', '-lxacc-fermion']
```
Comment (amccaskey) [inline tools/driver/qcor.in:14]: remove the merge text and just keep this line

File: runtime/qcor.hpp (hunk 4 above)
Comment (kharazit) [reply runtime/qcor.hpp:32]: This is what I thought too.

File: examples/fermi/fermi_op.cpp [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+      },
+      10);
+    auto results = optimizer->optimize(f);
+    printf("vqe energy = %f\n", results.first);
+    std::cout<<results.second[0]<<std::endl;
+    return 0;
+}
+
+
+/*
```
[the remark this answers is not in the document]
Comment (kharazit) [reply examples/fermi/fermi_op.cpp:66]: This was the equivalent fromString method for the above. Should I make an example with that method for fermions?

[the remark this answers is not in the document]
Comment (amccaskey) [reply examples/fermi/fermi_op.cpp:66]: no just leave the one showing your a adag functions

**Pull request merged**